openpype/hosts/resolve/api/lib.py

```diff
@@ -91,16 +91,41 @@ def get_current_project():
     return self.project_manager.GetCurrentProject()
 
 
-def get_current_timeline(new=False):
+def get_current_timeline(new=False, get_any=False):
+    """Get current timeline object.
+
+    Args:
+        new (bool, optional): return only new timeline. Defaults to False.
+        get_any (bool, optional): return any even new if no timeline available.
+            Defaults to False.
+
+    Returns:
+        object: resolve.Timeline
+    """
     # get current project
     project = get_current_project()
 
+    timeline = project.GetCurrentTimeline()
+
+    # return current timeline only if it is not new
+    if timeline and not new:
+        return timeline
+
+    # if get_any is True then return any timeline
+    if get_any:
+        timeline_count = project.GetTimelineCount()
+        if timeline_count == 0:
+            # if there is no timeline then create a new one
+            new = True
+        else:
+            return project.GetTimelineByIndex(1)
+
+    # create new timeline if new is True
     if new:
         media_pool = project.GetMediaPool()
         new_timeline = media_pool.CreateEmptyTimeline(self.pype_timeline_name)
         project.SetCurrentTimeline(new_timeline)
-
-    return project.GetCurrentTimeline()
+        return new_timeline
```

Comment:
I suppose this fixes the issue, but from a code perspective it seems a bit odd that `get_current_timeline` allows to get "any timeline" because when returned it's not the _current_ timeline? Should we also trigger `project.SetCurrentTimeline` for the any timeline case?

It feels weird that if `new` were True, but `get_any` was True that it wouldn't default to just returning the current timeline (if it did have a current timeline) since the function is called `get_current_timeline)`.

Also, it might be confusing that calling this function with `get_current_timeline(new=False, get_any=True)` still generates a new timeline. Which goes to show the `new` and `any` argument feel slightly confusing.

Admittedly I feel having dedicated functions `get_current_timeline()`, `get_new_timeline()`, `get_first_timeline()` might solve that and gain a lot in clarity.

Or `get_timeline(mode="current")` with also modes like "any_existing", "any_allow_new" and "new"`.

Reply:
Yeah, makes really sense to do it the way you recommend.